### ISSUE TITLE
Add Indexed access support to RealmResults

### ISSUE THREAD

**Issue by AndyDentFree** [realm/realm-dotnet#736]
From the SO question[Realm dotnet: get element by index from RealmResults](http://stackoverflow.com/questions/38593260/realm-dotnet-get-element-by-index-from-realmresults_) we need a way to provide indexed access.

After some discussion, we will first add the LINQ [ElementAt](https://msdn.microsoft.com/en-us/library/bb299233(v=vs.100).aspx) as seen in [LINQ source](
http://referencesource.microsoft.com/#System.Core/System/Linq/IQueryable.cs,aa8bee46a155ce96) and see if that satisfies the OP's needs.

**Comment by AndyDentFree** [comment]
Expressions like this in  `SimpleLINQtests.CreateList`  should work without the `ToList`

```
 var s0 = _realm.All<Person>().Where(p => p.Score == 42.42f),ToList();
 Assert.That(s0 [0].Score, Is.EqualTo(42.42f));
```

**Comment by kristiandupont** [comment]
This was resolved with #802.
